ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesfragment/valueobjects/LdesFragmentIdentifier.java

```diff
@@ -46,7 +46,7 @@ public static LdesFragmentIdentifier fromFragmentId(String fragmentId) {
 				List<FragmentPair> fragmentPairs = new ArrayList<>();
 				String[] fragmentPairStrings = splitString[1].split("&");
 				for (String fragmentPairString : fragmentPairStrings) {
-					String[] splitFragmentPairString = fragmentPairString.split("=");
+					String[] splitFragmentPairString = fragmentPairString.split("=", -1);
```

Comment:
why are you adding this '-1'?

Reply:
This is necessary to be have an empty string in the array. This happens in case of the substring fragmentation.